src/nwb_conversion_tools/datainterfaces/ecephys/spikeglx/spikeglxdatainterface.py

```diff
@@ -1,29 +1,47 @@
-"""Authors: Cody Baker and Ben Dichter."""
+"""Authors: Cody Baker, Heberto Mayorquin and Ben Dichter."""
 from datetime import datetime
 from pathlib import Path
 from typing import Optional
 
-from spikeextractors import SpikeGLXRecordingExtractor, SubRecordingExtractor, RecordingExtractor
+import spikeextractors as se
+from spikeinterface import BaseRecording
+from spikeinterface.extractors import SpikeGLXRecordingExtractor
+from spikeinterface.core.old_api_utils import OldToNewRecording
+
+
+from spikeextractors import SubRecordingExtractor, RecordingExtractor
 from pynwb.ecephys import ElectricalSeries
 
 from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
 from ..baselfpextractorinterface import BaseLFPExtractorInterface
 from ....utils import get_schema_from_method_signature, get_schema_from_hdmf_class, FilePathType
 
 
-def fetch_spikeglx_metadata(file_path: FilePathType, recording: RecordingExtractor, metadata: dict):
-    file_path = Path(file_path)
-    session_id = file_path.parent.stem
+def fetch_spikeglx_metadata(recording: BaseRecording, metadata: dict):
+    """
+    Fetches the session_start_time from the meta readings and adds it to the metadata.
+    """
 
-    if isinstance(recording, SubRecordingExtractor):
-        n_shanks = int(recording._parent_recording._meta.get("snsShankMap", [1, 1])[1])
-        session_start_time = datetime.fromisoformat(recording._parent_recording._meta["fileCreateTime"]).astimezone()
+    # Support for old spikeextractors objects
+    recording = recording._kwargs.get("oldapi_recording_extractor", recording)
+    if isinstance(recording, RecordingExtractor):
+        if isinstance(recording, SubRecordingExtractor):
+            meta = recording._parent_recording._meta
+        else:
+            meta = recording._meta
     else:
-        n_shanks = int(recording._meta.get("snsShankMap", [1, 1])[1])
-        session_start_time = datetime.fromisoformat(recording._meta["fileCreateTime"]).astimezone()
-    if n_shanks > 1:
+        stream_id = recording._kwargs["stream_id"]
+        meta = recording.neo_reader.signals_info_dict[(0, stream_id)]["meta"]
+
+    # These correspond to single shank channels https://billkarsh.github.io/SpikeGLX/help/imroTables/
+    imDatPrb_type = meta["imDatPrb_type"]
+    if imDatPrb_type not in ["0", "21"]:
         raise NotImplementedError("SpikeGLX metadata for more than a single shank is not yet supported.")
-    metadata["NWBFile"] = dict(session_start_time=session_start_time.strftime("%Y-%m-%dT%H:%M:%S"))
+
+    extracted_start_time = meta.get("fileCreateTime", None)
+    if extracted_start_time:
+        session_start_time = datetime.fromisoformat(extracted_start_time).astimezone().strftime("%Y-%m-%dT%H:%M:%S")
```

Comment:
Why are you converting this from a string to a datetime back to a string?

Reply:
My mistake, I also got this code from the previous implementation and did not stop to think about it. I just assumed that the metadata needed to be in a certain format and that this pipeline was doing it.

But as far as I remember the metadata needs to be in isoformat, correct? which the first element in the pipeline seems to assume anyway. Testing does confirm it can be removed.

Reply:
NWBFile can take a `datetime` object

Reply:
Right! which is what we did in the other PRs for this :)